modules/binary-tuple/src/main/java/org/apache/ignite/internal/binarytuple/inlineschema/TupleWithSchemaMarshalling.java

```diff
@@ -69,17 +70,21 @@ public final class TupleWithSchemaMarshalling {
         Object[] values = new Object[size];
         String[] columns = new String[size];
         ColumnType[] types = new ColumnType[size];
+        BitSet nestedTupleFlags = new BitSet();
 
         // Fill in the values, column names, and types.
         for (int i = 0; i < size; i++) {
             var value = tuple.value(i);
             values[i] = value;
             columns[i] = tuple.columnName(i);
             types[i] = inferType(value);
+            if (value instanceof Tuple) {
+                nestedTupleFlags.set(i);
+            }
```

Comment:
I think we should use a special column type id (`-1`) instead of a separate bit set - simpler and more compact.